src/com/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -288,9 +289,10 @@
     ;; confused if the database doesn't exist but we open and close a
     ;; connection without creating anything.
     (sql/with-connection write-db
-                         (scf-store/validate-database-version #(System/exit 1))
+                         (scf-store/validate-database-version
```

Comment:
if you use an external database you can run PE with an incompatible postgres.

Not sure I understand the second comment -- at the moment we don't process the config (except the tk hook) until we attempt to start the service, and the db validation is more or less the first thing we do. Are you talking about moving the validation into the let statement above under process-config, or moving it into config.clj? This comment gives me pause too FWIW: https://github.com/puppetlabs/puppetdb/blob/2.3.x/src/com/puppetlabs/puppetdb/cli/services.clj#L285-289